### ISSUE TITLE
Teku fails to recover after Internet connection is offline

### ISSUE THREAD

**Issue by dankrad** [Consensys-Incorporated/teku#2554]
### Description
After an internet connection drop, teku never came back online and re-synched. In fact, there are really concerning Errors in the log:
```02:53:00.363 ERROR - Validator   *** Produced invalid attestation for slot 47064 with expected validator index: 3598 and actual validator index: 21384 . Invalid reason: Aggregation bitlist size (135) does not match committee size (134)
```
and
```02:16:00.379 ERROR - Validator   *** Failed to produce attestation  Slot: 46879 tech.pegasys.teku.core.signatures.SlashableConditionException: Refusing to sign attestation at slot 46879 with source epoch 1461 and target epoch 1464 because it may violate a slashing condition (See log file for full stack trace)
```

### Acceptance Criteria
Teku should automatically resume operations on connection drops. Validators should not produce invalid attestations.

### Steps to Reproduce (Bug)

**Expected behavior:** [What you expect to happen]

**Actual behavior:** [What actually happens]

**Frequency:** [What percentage of the time does it occur?]

### Versions (Add all that apply)
* Software version: teku/v0.12.3-dev-2993d20d/linux-x86_64/-ubuntu-openjdk64bitservervm-java-11
* Java version: openjdk version "11.0.8" 2020-07-14
* OS Name & Version: Ubuntu 20.04 LTS
* Kernel Version: Linux pascal 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
* Virtual Machine software & version: /
* Docker Version: /
* Cloud VM, type, size: /

**Comment by ajsutton** [comment]
Hi @dankrad The exceptions didn't come through.  We have had issues in the past with connections not restoring but recent tests have been good so would be very useful to see the content of those exceptions and if at all possible the full log output.

**Comment by dankrad** [comment]
[tekulog.zip](https://github.com/PegaSysEng/teku/files/5056493/tekulog.zip)

**Comment by ajsutton** [comment]
Thanks. It's getting late here so I won't be able to look into this much now but we'll investigate.

The messages:
```
2020-08-11 11:29:28.634+01:00 | P2PService-event-thread-0 | WARN  | AbstractRouter | AbstractRouter internal error on message null from peer null
io.libp2p.core.InternalErrorException: [peerHandler] not initialized yet
```
are harmless (albeit very annoying) and will be fixed by https://github.com/PegaSysEng/teku/pull/2549

I haven't had a chance to look into the other messages yet.

**Comment by mbaxter** [comment]
It looks like the node gets stuck in a way that looks very similar to the caching issue described [here](https://github.com/PegaSysEng/teku/pull/2501)
```
2020-08-11 02:14:00.374+01:00 | validator-async-19127 | INFO  | teku-validator-log | [34mValidator   *** Published attestation  Count: 4, Slot: 46869, Root: 09bff2..82b0[0m

2020-08-11 02:14:08.344+01:00 | TimeTickChannel-0 | INFO  | teku-event-log | [37mSync Event  *** Current slot: 46870, Head slot: 46723, Connected peers: 2[0m

2020-08-11 02:14:32.353+01:00 | TimeTickChannel-0 | INFO  | teku-event-log | [37mSync Event  *** Current slot: 46872, Head slot: 46781, Connected peers: 2[0m

2020-08-11 02:14:48.400+01:00 | validator-async-19125 | INFO  | teku-validator-log | [34mValidator   *** Published attestation  Count: 2, Slot: 46873, Root: 66506c..c73b[0m

2020-08-11 02:14:50.352+01:00 | TimeTickChannel-0 | INFO  | teku-event-log | [37mSync Event  *** Current slot: 46873, Head slot: 46781, Connected peers: 1[0m

2020-08-11 02:15:40.466+01:00 | p2p-async-154 | WARN  | PeerSync | Failed to import block from peer (err: FAILED_STATE_TRANSITION) SignedBeaconBlock{message=BeaconBlock{root=0x0b2e30dfba428e51f1d21222a6238bcf70a45548086b51e98ef38afe27baeb22, slot=46787, proposer_index=13778, parent_root=0x39fbe8ebbd97a8bbd952000639934796d80b5f44e91127b84e57dc41bc8caa50, state_root=0xbb5a093c3f9a54f6b56dddd215b81ddacdc39529ef2e4f5b26446f67a4fcd966, body=0x8bf4ed333bffd9544083f0cf0caeb8e0388f9933c1cab9cb08034e3f5bd956e7}, signature=0x9299f97327834c7c61df786bff1ab2acf97c70ff80efa62393ccf24220d2e165795b94fb2eaae0fbaf540788706d1b4a054940dc33fe521e3b445a4c951aa9faba555654964de2290cefdeb9de414f6fc07e3843ac32337e04ecc474f18c0c13}: Eth2Peer{id=16Uiu2HAm2i5u6hKcANdRomyYQz1bq1syx7EhGDratyz5TLsnA86E, remoteStatus=Optional[PeerStatus{forkDigest=0xe7a75d5a, finalizedRoot=0x39fbe8ebbd97a8bbd952000639934796d80b5f44e91127b84e57dc41bc8caa50, finalizedEpoch=1462, headRoot=0x1339768c2aafc495962c89370d56c03ef85a25271ee1de5e4a95112ea9025340, headSlot=46876}]}
```

The chain head has fallen behind because of connectivity issues, we go to produce an attestation for the current slot, using an old chain head.  Because of a bug in our caching / attestation production logic (now fixed), we would cache the committee information calculated from this old state.  When we go to import a new block, we pull invalid committee information from our cache, causing us to see invalid attestations in the block we're importing.  Since the cache is now corrupted, the node can't recover and import any new blocks.

Try updating to the latest code from master, which has a fix for this issue.

**Comment by mbaxter** [comment]
Closing this for now - as this should be addressed by the fix to #2446

Please reopen if you run into this again.
